### PR TITLE
protocols/ping: Properly deprecate types with `Ping` prefix

### PR DESCRIPTION
**Pull request by thomaseizinger**
# Description

The deprecation we tried to do in https://github.com/libp2p/rust-libp2p/pull/2215 doesn't work but defining type aliases works :)

## Links to any relevant issues

- https://github.com/libp2p/rust-libp2p/issues/2217
- https://github.com/libp2p/rust-libp2p/pull/2215

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] ~I have made corresponding changes to the documentation~
- [ ] ~I have added tests that prove my fix is effective or that my feature works~
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Review by jxs — COMMENTED** [review]
Hi Thomas, when doing #2927 noticed there are still some references to the deprecated types  in the docs, and some types with `Ping` prefix, should they be addressed?

(sorry for doing it like this, github still doesn't support [reviewing on unchanged files nor unchanged code](https://github.com/github/roadmap/issues/347))

**docs:**

https://github.com/libp2p/rust-libp2p/blob/a772e5ae17c0963873f76bb8cd65315f7768400b/protocols/ping/src/lib.rs#L29

https://github.com/libp2p/rust-libp2p/blob/a772e5ae17c0963873f76bb8cd65315f7768400b/protocols/ping/src/lib.rs#L34

https://github.com/libp2p/rust-libp2p/blob/a772e5ae17c0963873f76bb8cd65315f7768400b/protocols/ping/src/lib.rs#L38

https://github.com/libp2p/rust-libp2p/blob/a772e5ae17c0963873f76bb8cd65315f7768400b/protocols/ping/src/handler.rs#L59

https://github.com/libp2p/rust-libp2p/blob/a772e5ae17c0963873f76bb8cd65315f7768400b/protocols/ping/src/handler.rs#L211

**structs:**

https://github.com/libp2p/rust-libp2p/blob/a772e5ae17c0963873f76bb8cd65315f7768400b/protocols/ping/src/handler.rs#L388

https://github.com/libp2p/rust-libp2p/blob/a772e5ae17c0963873f76bb8cd65315f7768400b/protocols/ping/src/handler.rs#L384

**Comment by elenaf9** [comment]
What do you think about just removing the deprecated types? They have been marked as deprecated for quite some time now, and upgrading to the new names should be easy.

**Comment by thomaseizinger** [comment]
> What do you think about just removing the deprecated types? They have been marked as deprecated for quite some time now, and upgrading to the new names should be easy.

They have been incorrectly marked as deprecated :)

**Comment by elenaf9** [comment]
> > What do you think about just removing the deprecated types? They have been marked as deprecated for quite some time now, and upgrading to the new names should be easy.
> 
> They have been incorrectly marked as deprecated :)

Incorrectly in the sense that users that were still using them did not get a warning? Does the `deprecated` macro not work on re-exports?

File: protocols/ping/CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,11 @@
 # 0.39.1 [unreleased]
 
 - Bump rand to 0.8 and quickcheck to 1. See [PR 2857].
+- Deprecate types with `Ping` prefix. Prefer importing them via the `ping` namespace, i.e. `libp2p::ping::Event` instead
+  of `libp2p::ping::PingEvent`. See [PR XXXX].
 
 [PR 2857]: https://github.com/libp2p/rust-libp2p/pull/2857
+[PR XXXX]: https://github.com/libp2p/rust-libp2p/pull/XXXX
```
Comment (elenaf9) [inline protocols/ping/CHANGELOG.md]:
```suggestion
- Deprecate types with `Ping` prefix. Prefer importing them via the `ping` namespace, i.e. `libp2p::ping::Event` instead
  of `libp2p::ping::PingEvent`. See [PR 2937].

[PR 2857]: https://github.com/libp2p/rust-libp2p/pull/2857
[PR 2937]: https://github.com/libp2p/rust-libp2p/pull/2937
```

**Comment by jxs** [comment]
> > > What do you think about just removing the deprecated types? They have been marked as deprecated for quite some time now, and upgrading to the new names should be easy.
> > 
> > 
> > They have been incorrectly marked as deprecated :)
> 
> Incorrectly in the sense that users that were still using them did not get a warning? Does the `deprecated` macro not work on re-exports?

Hi Elena, yeah exactly. See https://github.com/libp2p/rust-libp2p/pull/2927#discussion_r977392442

**Review by mxinden — APPROVED** [review]
Looks good to me. Thanks for making it easier for users to upgrade.

If I am not mistaken, this is missing the suggestions by @jxs. Otherwise looks good to me.

**Comment by thomaseizinger** [comment]
> If I am not mistaken, this is missing the suggestions by @jxs. Otherwise looks good to me.

Yep! Unfortunately, rustdoc doesn't flag links to deprecated items as deprecated. Grml.

**Comment by thomaseizinger** [comment]
Okay, I went across the repo again with fulltext search and found quite a few more usages.

I also realised that our CI doesn't check for warnings in examples! I'll see to fix that too.

**Comment by thomaseizinger** [comment]
> I also realised that our CI doesn't check for warnings in examples! I'll see to fix that too.

Done: https://github.com/libp2p/rust-libp2p/pull/2949

**Review by jxs — COMMENTED** [review]
> Okay, I went across the repo again with fulltext search and found quite a few more usages.
> 
> I also realised that our CI doesn't check for warnings in examples! I'll see to fix that too.

awesome, LGTM! 
Did you see the references to `PingState` and `PingFuture` ? `PingState` can be renamed to just `State` as it isn't shared outside of the module. though `PingFuture` and `PongFuture` probably there isn't a better wording for them.

**Comment by thomaseizinger** [comment]
> > Okay, I went across the repo again with fulltext search and found quite a few more usages.
> > I also realised that our CI doesn't check for warnings in examples! I'll see to fix that too.
> 
> awesome, LGTM! Did you see the references to `PingState` and `PingFuture` ? `PingState` can be renamed to just `State` as it isn't shared outside of the module. though `PingFuture` and `PongFuture` probably there isn't a better wording for them.

I guess `PingState` could be renamed yeah. I think public API types are the most important. Everything else we can any time :)

**Review by elenaf9 — APPROVED** [review]

**Review by jxs — APPROVED** [review]

**Pull request merged**